client/extensions/hello-dolly/state/reducer.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Internal dependencies
+ */
+import {
+	ROUTE_SET,
+	SECTION_SET,
+	SITE_SETTINGS_SAVE,
+} from 'state/action-types.js';
+
+export default function( state = 0, action ) {
+	switch ( action.type ) {
+		case ROUTE_SET:
+		case SECTION_SET:
+		case SITE_SETTINGS_SAVE:
+			return advanceToNextLyric( state );
+		default:
+			return state;
+	}
+}
+
+function advanceToNextLyric( index ) {
+	return index + 1;
+}
```

Comment:
@coderkevin I think part of the reason you feel like keeping this function makes the intent more clear could be related to the fact that the reducer has no name. `advanceToNextLyric()` seems like a function which updates some internal reference to a lyric _but this reducer itself is that reference_.

while I won't press you to reconsider, I might suggest that you give the reducer a fitting name. that in itself might clarify the logic enough and maybe you wouldn't see the same demand for this function.

let me rephrase: the reducer isn't a function which is responsible for all of the lyric state - it's a simple function governing a tiny piece of primitive data. _any_ and _all_ reducers governing the state of some system or subsystem are _a composition of little reducers_.

```js
export const lyricIndex = ( state = 0, { type } ) => { … };

// or in the style you've written…

export function lyricIndex( state = 0, { type } ) {
	…
}
```

for now that's good enough to encapsulate all state. if we had a need for more or we knew we would expand it in the future we use `combineReducers()`…

```js
export default combineReducers( {
	lyricIndex,
} );
```